ablySpec/RealtimeClientChannel.swift

```diff
@@ -769,6 +876,31 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                it("with a attach callback should subscribe and call the callback when attached") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+
+                    let publishedMessage = ARTMessage(name: "foo", data: "bar")
+
+                    waitUntil(timeout: testTimeout) { done in
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Initialised))
+
+                        channel.subscribeWithAttachCallback({ errorInfo in
```

Comment:
Does this method need to be overloaded given we've "preferred" to stick with the overloaded IDL?